src/main/java/org/example/util/UserUtil.java

```diff
@@ -0,0 +1,67 @@
+package org.example.util;
+
+import lombok.experimental.UtilityClass;
+import org.example.models.Trainee;
+import org.example.models.Trainer;
+import org.example.models.Training;
+import org.example.service.TraineeService;
+import org.example.service.TrainerService;
+import org.example.service.TrainingService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+public class UserUtil {
```

Comment:
I'd rename it to UserService. Usually, util classes are small and have only static methods